### ISSUE TITLE
Activities type=COMMENTED does not contain the comment

### ISSUE THREAD

**Issue by wiibaa** [cdancy/bitbucket-rest#102]
Me again :),

actually starting to write my client, I stumble upon how to list the general comments of a PR.

Desperately trying to use GET pullRequest/comments/ I finally found out  that it is not the way to go from https://community.atlassian.com/t5/Answers-Developer-Questions/Reading-pull-request-comments-via-REST-API-fails/qaq-p/523952 confirmed by https://jira.atlassian.com/browse/BSERV-7643

that pointed me to the Activities of a PR thus calling **listActivities()** but the Activities object do not contains the comment related fields `commentAction`,`comment` and `commentAnchor`

help please :smiley:

**Comment by cdancy** [comment]
I'll have a look at this one tomorrow morning after working on the tasks impl for you. Stay tuned!!!

**Comment by cdancy** [comment]
If you're looking for all the comments of a given PR you can iterate over each page through this endpoint:

https://github.com/cdancy/bitbucket-rest/blob/master/src/main/java/com/cdancy/bitbucket/rest/features/CommentsApi.java#L93

**Comment by wiibaa** [comment]
I stay tuned for sure.
THe issue with the CommentsApi is that the path parameter is mandatory, so it makes it difficult to iterate over comments directly without knowing the set of files that are modified.
Unless you implemented some extra logic to ease the loop path is a required param and do not accept a wildcard

````{
"errors": [
{
"context": null,
"message": "The path query parameter is required when retrieving comments.",
"exceptionName": null
}
]
}
````

**Comment by wiibaa** [comment]
Forgot to post an example, this is the kind of activity I would like to search for

````
{
	"id": 275724,
	"createdDate": 1499150545790,
	"user": {
		"name": "xxx",
		"emailAddress": "xxx",
		"id": 18325,
		"displayName": "xxx",
		"active": true,
		"slug": "xxx",
		"type": "NORMAL",
		"links": {
			"self": [{
				"href": "https://xxx/users/xxx"
			}]
		}
	},
	"action": "COMMENTED",
	"commentAction": "ADDED",
	"comment": {
		"properties": {
			"repositoryId": 183
		},
		"id": 15900,
		"version": 0,
		"text": "Comment text markdown",
		"author": {
			"name": "xxx",
			"emailAddress": "xxx",
			"id": 18325,
			"displayName": "xxx",
			"active": true,
			"slug": "xxx",
			"type": "NORMAL",
			"links": {
				"self": [{
					"href": "https://xxx/users/xxx"
				}]
			}
		},
		"createdDate": 1499150545790,
		"updatedDate": 1499150545790,
		"comments": [],
		"tasks": [],
		"permittedOperations": {
			"editable": false,
			"deletable": true
		}
	}
}
```

**Comment by cdancy** [comment]
@wiibaa I've got things wired up for the above endpoint to work as you would expect. Were you looking to automate the posting of tasks as well or just querying for them?

**Comment by wiibaa** [comment]
Found another issues about listing activities

````
Expected a string but was BEGIN_OBJECT at line 1 column 18733 path $.values[7].comment.properties.
````

Caused by the `likedBy` property
````
"comment": {
  "properties": {
    "repositoryId": 183,
    "likedBy": {
      "total": 1,
      "likers": [
        {
           "name": "xxx",
           "emailAddress": "xxx",
           "id": 8447,
           "displayName": "WEBER Philippe",
           "active": true,
           "slug": "xxx",
           "type": "NORMAL",
           "links": {
             "self": [
               {
               "href": "https://xxx/stash/users/xxx"
               }
             ]
          } 
        }
      ]
    }
  },
````

**Comment by cdancy** [comment]
@wiibaa hmmm ... that is no good IMO. Are you on the latest and greatest version of Bitbucket? If so this seems sloppy on their part to change the properties from a generic String/String map to String/Object map. Poking around online now and can't find any documentation on this.

**Comment by wiibaa** [comment]
@cdancy not the very latest but at least on the 5.0 branch:
`Version{version=5.0.2, buildNumber=5000002, buildDate=1496198334146, displayName=Bitbucket}`
from the release log its from May 2017

**Comment by cdancy** [comment]
If you can find any documentation on it that would be great. I can't seem to find anything.

If however they are going to be doing String/Object for the `properties` map then the best I could do here would be to return a String/Object map or a String/JSONElement map. With the former the user would have to cast it to "something" which I don't think makes much sense while the latter the user could iterate over the JSONElement and pick out what he/she is interested in. Let me think about this for a bit to see if there are any better solutions.

**Comment by wiibaa** [comment]
also looking around its seems the attributes(4.x) and properties (5.x) can be used by plugins so indeed seems to be rather free from my understanding of https://jira.atlassian.com/browse/BSERV-7876?focusedCommentId=807932&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-807932

**Comment by cdancy** [comment]
Think you're right here. Let me put in quick fix and we'll see how things fare. Stay tuned...

**Comment by cdancy** [comment]
@wiibaa just put in a fix. Give things a go and let me know how it works out. All tests pass here so I think we should be good to go.

**Comment by wiibaa** [comment]
Run the scan on all the PR of one of our big repo, and no hiccup.
Did not expected to have this much impact on existing code but everything should be smooth from now thanks to you.

**Comment by cdancy** [comment]
No problem. Plan on doing another release this weekend so the official version should be out soon.

**Comment by cdancy** [comment]
Closing as this functionality was merged in #106
